phpstan.neon.dist

```diff
@@ -0,0 +1,17 @@
+parameters:
+    level: 3
+    paths:
+        - lib
+    ignoreErrors:
+        # Making classes final as suggested would be a BC-break
+        -
+            message: '~Unsafe usage of new static\(\)\.~'
+            paths:
+                - 'lib/Doctrine/Common/Collections/ArrayCollection.php'
+                - 'lib/Doctrine/Common/Collections/Criteria.php'
+        -
+            message: '~Array \(array\<TKey, T\>\) does not accept key int\.~'
+            path: 'lib/Doctrine/Common/Collections/ArrayCollection.php'
+        -
+            message: '~Method Doctrine\\Common\\Collections\\ArrayCollection\:\:map\(\) should return Doctrine\\Common\\Collections\\ArrayCollection\<TKey, U\> but returns Doctrine\\Common\\Collections\\ArrayCollection\<TKey, T\>\.~'
```

Comment:
This one should not be suppressed, but rather fixed in `createFrom()` docblock. The following needs to be added/changed:
```
     * @psalm-template K of array-key
     * @psalm-template V
     * @phpstan-template K
     *
     * @psalm-param array<K,V> $elements
     * @psalm-return static<K,V>
```